### PR TITLE
Missions

### LINKED ISSUE #15: [4pt] Display missions - Lists render
- Use `useSelector()` Redux Hook to select the state slices and render lists of  missions in corresponding routes. i.e.:
```javascript
// get rockets data from the store
const rockets = useSelector(state => state.rockets);
```

- You can style the whole application "by hand" or you could use [React Bootstrap](https://react-bootstrap.github.io/), a UI library that could speed up the process. This is a popular library and working with its components would be good practice.
- Render a table with the missions' data (as per design).

### LINKED ISSUE #18: [4pt]  Fetch missions - Fetch data
Fetch data from the Missions endpoint  (https://api.spacexdata.com/v3/missions) when a user navigates to the Missions section.

Once the data are fetched, dispatch an action to store the selected data in Redux store:

- mission_id
- mission_name
- description

NOTE: Make sure you only dispatch those actions once and do not add data to store on every re-render (i.e. when changing views / using navigation).

### PR DESCRIPTION
**Pull request by rustam-tolipov**
**Missions: Fetch missions, Display missions**

All requirements are complete: 

- Fetching data from API 💯 
- The mission page is displayed and styled 💯 
- No linters errors 💯 

Feel free to review -🧑🏻‍💻  - my project and let me know if there are any changes to do. 🤓 -😉

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
 
 import './App.css';
 
 function App() {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchPostsRequestMissions());
+  }, []);
+
+  const missions = useSelector((state) => state.spaceReducer.missions);
```
Comment (ahzia) [inline src/App.js]: Required: Move this to the mission component.
The data should be only fetched when a user visits the mission component

File: src/redux/missions/middlewares.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import logger from 'redux-logger';
+import spaceReducer from './missions';
+
+const reducer = combineReducers({
+  spaceReducer,
+});
+
+const store = createStore(reducer, applyMiddleware(thunk, logger));
+
+export default store;
```
Comment (ahzia) [inline src/redux/missions/middlewares.js]: Move this code to configureStore.js

File: src/redux/missions/missions.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+const initialState = {
+  loading: true,
+  missions: [],
+  userMissions: [],
+};
+
+export const fetchPostsRequestMissions = () => async (dispatch) => {
+  dispatch(fetchPostsLoading());
+  const request = await fetch('https://api.spacexdata.com/v3/missions');
+  const result = await request.json();
```
Comment (ahzia) [inline src/redux/missions/missions.js]: Move function related to API to /API/spacedata.js and imported back anywhere you need it.
It's better, all our API request functions be in a specific file.

File: src/redux/missions/missions.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+          mission_id,
+          mission_name,
+          description,
+          join: false,
+        }))(mission);
+        return selectedData;
+      })
+    )
+  );
+};
```
Comment (ahzia) [inline src/redux/missions/missions.js]: Optional: Move the Middleware functions to middleware.js and import them back where you need them.
It's better to spread the main reducer file and middleware so our code will be more readable.

File: src/redux/missions/middlewares.js (hunk 2 above)
Comment (rustam-tolipov) [reply src/redux/missions/middlewares.js]: This problem has been resolved.

**Review by ahzia — CHANGES_REQUESTED** [review]
Hi @Rustamxon7,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Required Changes ♻️
- [x] You have used too much ESLint diable rule, instead fix the linters issue in your code.
- [x] Check the Inline Comments
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Review by ahzia — APPROVED** [review]
# Hi @Rustamxon7 , 

### Status: Approved ✔️

:+1: You have implemented all the suggestions made in the previous review. Well Done.

Your project is complete!  There are some changes and refactoring required and I will commit it, other than that nothing else to say.
Congratulations! 🎉 

**You may close this Pull Request and merge the changes**
  
**Keep being awesome devs and good luck with your next milestone!**
-----

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**